### ISSUE TITLE
Add Moonshine as transcription backend option

### ISSUE THREAD

**Issue by jdcockrill** [jdcockrill/voxvibe#91]
## Overview

Add support for [Moonshine](https://github.com/moonshine-ai/moonshine) as an alternative transcription backend alongside faster-whisper and Voxtral.

## What is Moonshine?

Moonshine is a family of speech-to-text models released by Useful Sensors in October 2024, specifically optimized for real-time transcription on resource-constrained devices. It's particularly well-suited for VoxVibe's use case of short-form voice dictation.

## Why Integrate Moonshine?

1. **Speed**: 5-15x faster than Whisper on short audio clips (10-30 seconds)
2. **Efficiency**: Compute scales with audio length, unlike Whisper which always processes 30-second chunks
3. **Size**: Tiny model is only 27M params (190MB), Base is 62M params (400MB)
4. **Accuracy**: Better WER than Whisper Tiny and Base despite smaller size
5. **License**: MIT (very permissive)
6. **Multilingual**: Supports English, Arabic, Chinese, Japanese, Korean, Ukrainian, Vietnamese (2025 variants)

## Technical Details

- **Python Package**: `useful-moonshine` available on PyPI
- **Framework**: Uses Keras with PyTorch/TensorFlow/JAX backend support
- **Architecture**: Encoder-decoder transformer with Rotary Position Embedding (RoPE)
- **Models**: 
  - `moonshine/tiny` - 27M parameters
  - `moonshine/base` - 62M parameters
  - Multilingual variants available

## Implementation Tasks

- [ ] Add `useful-moonshine` dependency to `pyproject.toml`
- [ ] Create `app/voxvibe/transcription/moonshine_transcriber.py` implementing the transcriber interface
- [ ] Add `backend = "moonshine"` configuration option to `config.toml`
- [ ] Update configuration validation to support moonshine backend
- [ ] Add model selection (tiny/base) to configuration
- [ ] Update documentation (README.md, CLAUDE.md) with moonshine option
- [ ] Test integration with VoxVibe's audio pipeline
- [ ] Compare performance vs faster-whisper for typical VoxVibe usage patterns

## Configuration Example

```toml
[transcription]
backend = "moonshine"  # Options: "faster-whisper", "voxtral", "moonshine"
model = "base"  # For moonshine: "tiny" or "base"
```

## References

- [Moonshine GitHub Repository](https://github.com/moonshine-ai/moonshine)
- [Moonshine ArXiv Paper](https://arxiv.org/abs/2410.15608)
- [useful-moonshine on PyPI](https://pypi.org/project/useful-moonshine/)
- [Performance Comparison Analysis](https://northflank.com/blog/best-open-source-speech-to-text-stt-model-in-2026-benchmarks)
- [The Decoder: Moonshine is 5x faster than Whisper](https://the-decoder.com/open-source-moonshine-speech-recognition-model-is-up-to-five-times-faster-than-openais-whisper/)

## Notes

Since VoxVibe users typically dictate short bursts of text (5-30 seconds), Moonshine's optimizations for short audio clips could significantly improve the perceived responsiveness of the application.

**Comment by m13v** [comment]
moonshine is genuinely impressive for short audio. we benchmarked it against whisper small and the latency difference on sub-30-second clips was dramatic, something like 3-4x faster on CPU. the catch is accuracy on proper nouns and technical terms. whisper base consistently got names and domain-specific vocab right where moonshine tiny would hallucinate. if your users dictate a lot of names or jargon, i'd default to moonshine base (62M params) instead of tiny. the file size difference is small but the accuracy gap is real. also worth noting moonshine's multilingual variants are newer and less tested than whisper's, so if you support non-English users you might want to keep faster-whisper as the default for those languages.

**Comment by m13v** [comment]
our WhisperKit transcription service with model selection, confidence-based filtering, and the full processing pipeline: https://github.com/m13v/fazm/blob/main/Desktop/Sources/TranscriptionService.swift

the model management piece (download, caching, loading states) is probably the most reusable part if you're adding moonshine alongside existing backends.
